go/vt/vtctl/workflow/server.go

```diff
@@ -4345,6 +4358,50 @@ func (s *Server) mirrorTraffic(ctx context.Context, req *vtctldatapb.WorkflowMir
 	return nil
 }
 
+// validateShardsHaveVReplicationPermissions checks that the primary tablets
+// in the given keyspace shards have the required permissions necessary to
+// perform actions on the workflow.
+func (s *Server) validateShardsHaveVReplicationPermissions(ctx context.Context, keyspace string, shards []*topo.ShardInfo) error {
+	var wg sync.WaitGroup
+	allErrors := &concurrency.AllErrorRecorder{}
```

Comment:
Tip of the day: I've recently been introduced to `errgroup.WithContext(...)` which is like `WaitGroup` but also has the builtin mechanism to aggregate the errors, and also auto-terminates all goroutines when the context expires. I've started using it recently and am very pleased with it.

Reply:
I've used it quite a few times, but for whatever reasons did not think about using it here (I think because I started with a cut and paste of a similar func). I'll switch to that. Thanks for the reminder!

Reply:
Changed in: https://github.com/vitessio/vitess/pull/16762/commits/6ec4b72773a359aff0d57c197bca53c88ae92b06